docs/function.rst

```diff
@@ -0,0 +1,147 @@
+.. _function:
+
+Function
+========
+
+Melody functions are user-written Python functions that perform the
```

Comment:
The term Objective function is often used. I've re-read the function section and I think it reads OK :-). One issue is that a Melody function is written as a Python function so the word function is used a lot. I've made one minor change and will commit and push hoping you either don't notice I've not really addressed your point or have a change of heart.